ecosystem/python/sdk/aptos_sdk/transaction_worker.py

```diff
@@ -0,0 +1,225 @@
+# Copyright © Aptos Foundation
+# SPDX-License-Identifier: Apache-2.0
+
+import asyncio
+import logging
+import typing
+
+from aptos_sdk.account import Account
+from aptos_sdk.account_address import AccountAddress
+from aptos_sdk.account_sequence_number import AccountSequenceNumber
+from aptos_sdk.async_client import RestClient
+from aptos_sdk.transactions import SignedTransaction, TransactionPayload
+
+
+class TransactionWorker:
+    """
+    The TransactionWorker provides a simple framework for receiving payloads to be processed. It
+    acquires new sequence numbers and calls into the callback to produce a signed transaction, and
+    then submits the transaction. In another task, it waits for resolution of the submission
+    process or get pre-execution validation error.
+
+    Note: This is not a particularly robust solution, as it lacks any framework to handle failed
+    transactions with functionality like retries or checking whether the framework is online.
+    This is the responsibility of a higher-level framework.
+    """
+
+    _account: Account
+    _account_sequence_number: AccountSequenceNumber
+    _rest_client: RestClient
+    _transaction_generator: typing.Callable[
+        [Account, int], typing.Awaitable[SignedTransaction]
+    ]
+    _started: bool
+    _stopped: bool
+    _outstanding_transactions: asyncio.Queue
+    _outstanding_transactions_task: typing.Optional[asyncio.Task]
+    _processed_transactions: asyncio.Queue
+    _process_transactions_task: typing.Optional[asyncio.Task]
+
+    def __init__(
+        self,
+        account: Account,
+        rest_client: RestClient,
+        transaction_generator: typing.Callable[
+            [Account, int], typing.Awaitable[SignedTransaction]
+        ],
+    ):
+        self._account = account
+        self._account_sequence_number = AccountSequenceNumber(
+            rest_client, account.address()
+        )
+        self._rest_client = rest_client
+        self._transaction_generator = transaction_generator
+
+        self._started = False
+        self._stopped = False
+        self._outstanding_transactions = asyncio.Queue()
+        self._processed_transactions = asyncio.Queue()
+
+    def account(self) -> AccountAddress:
+        return self._account.address()
+
+    async def _submit_transactions_task(self):
+        try:
+            while True:
+                sequence_number = (
+                    await self._account_sequence_number.next_sequence_number()
+                )
+                transaction = await self._transaction_generator(
+                    self._account, sequence_number
+                )
+                txn_hash_awaitable = self._rest_client.submit_bcs_transaction(
+                    transaction
+                )
```

Comment:
just a thought (not sure you want to overcomplicate this) - one potential fault-tolerant improvement, is to have AccountSequenceNumber have two counters, issued and submitted, and next_sequence_number increments issued, and after submit_bcs_transaction is called , if transaction was potentially submitted (we received success response, or received no response) we notify AccountSequenceNumber to increment submitted counter
If we received failed response for submission, we notify AccountSequenceNumber to decrement issued number.

and AccountSequenceNumber is changed to never have issued be more than 1 above submitted. 
that would make it such that single failure in submission doesn't look up the whole stream until expiry